### PR TITLE
PR SdV 2 - Errors and performance.

### PR DESCRIPTION
**Pull request by sdv-at-3e-eu**
Correct two errors after introducing their test cases.
Add some code simplification with performance benefits.

(This PR builds up on PR SdV -1; I hope it will be easy to ignore the already processed commits.

### CODE REVIEW

File: src/local-time.lisp [hunk 1]
```diff
@@ -197,28 +197,19 @@
 
 (defun transition-position (needle haystack)
```
Comment (dlowe-net) [inline src/local-time.lisp:198]: A much tighter implementation. Thanks.

File: src/local-time.lisp [hunk 2]
```diff
@@ -541,17 +541,9 @@ In other words:
...
-    (let* ((new-sec (+ sec offset-sec))
-           (new-day (+ day offset-day)))
-      (cond ((minusp new-sec)
-             (incf new-sec +seconds-per-day+)
-             (decf new-day))
-            ((>= new-sec +seconds-per-day+)
-             (incf new-day)
-             (decf new-sec +seconds-per-day+)))
-      (values new-sec new-day))))
+  (multiple-value-bind (offset-day new-sec)
```
Comment (dlowe-net) [inline src/local-time.lisp:544]: Yeah, this is way better.

File: src/local-time.lisp [hunk 3]
```diff
@@ -1738,16 +1738,16 @@ See the documentation of FORMAT-TIMESTRING for the structure of FORMAT."
...
                  (if (and (eql fmt :gmt-offset-or-z) (zerop offset))
                      (princ #\Z result)
                      (format result "~c~2,'0d~:[:~;~]~2,'0d"
-                             (if (minusp offset-hours) #\- #\+)
+                             (if (minusp offset) #\- #\+)
                              (abs offset-hours)
                              (eql fmt :gmt-offset-hhmm)
-                             (truncate (abs offset-secs)
-                                       +seconds-per-minute+)))))
+                             (round (abs offset-secs)
```
Comment (dlowe-net) [inline src/local-time.lisp:1749]: I'm concerned this will change output in a breaking manner.

Comment (sdv-at-3e-eu) [reply src/local-time.lisp:1749]: It will, although only for offsets that do not even align on minutes.
I'll revert.

**Pull request merged**